pkg/core/native/native_neo.go

```diff
@@ -437,6 +463,21 @@ func (n *NEO) PostPersist(ic *interop.Context) error {
 			}
 		}
 	}
+	// Update next block validators cache for external users.
+	var (
+		h        = ic.Block.Index // consider persisting block as stored to get _next_ block validators
+		numOfCNs = n.cfg.GetNumOfCNs(h + 1)
+	)
+	if cache.validators == nil || numOfCNs != len(cache.validators) {
```

Comment:
Can be reduced to `numOfCNs != len(cache.validators)`, but it's a little hackish...

Reply:
It can, but could we keep it as is to avoid misunderstanding in future if one day we'll extend/refactor this code? To be honest, I'm afraid of touching NEO cache.

Reply:
I'm sure that `numOfCNs != len(cache.validators)` won't change the things, but current version may help to understand what's going on in other contract parts.